File: js/desafio1.js
const a = 60;
let divisores = [];
function obtenerDivisores(a) {
    for (let i = a; i > 0; i--) {
        if (a % i === 0) {
            divisores.push(i);
        }
    }
}

obtenerDivisores(a); //60, 30, 20, 15, 12, 10, 6, 5, 4, 3, 2, 1

function cadenas(a) {
    let suma = [];

    for (let i = 0; i < divisores.length; i++) {
        if (a % divisores[i] === 0) {
            suma.push(divisores[i]);
            a = divisores[i];
        }
    }

    return suma.reduce((n1, n2) => n1 + n2);
}

console.log(cadenas(a)); //resultado: 60+30+15+5+1 (111)

File: c++/desafio1.cpp
#include <iostream>
#include <vector>

const int a = 60;

std::vector<int> divisores;

void obtenerDivisores(int num) {
    for (int i = num; i > 0; i--) {
        if (num % i == 0) {
            divisores.push_back(i);
        }
    }
}

int cadenas(int a) {
    std::vector<int> suma;
    int res = 0;

    for (int i = 0; i < divisores.size(); i++) {
        if (a % divisores[i] == 0) {
            suma.push_back(divisores[i]);
            a = divisores[i];
        }
    }

    for (int num : suma) {
        res += num;
    }

    return res;
}

int main() {
    obtenerDivisores(a);

    std::cout << cadenas(a); //Resultado: 60 + 30 + 15 + 5 + 1 = 111

    return 0;
}